### PR TITLE
bugfix: resolution when apisix reload eureka does not load the data

### PR DESCRIPTION
**Pull request by kongjun01**
### Description

https://github.com/apache/apisix/issues/7178

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/discovery/eureka/init.lua [hunk 1]
```diff
@@ -16,15 +16,14 @@
 --
 
 local local_conf         = require("apisix.core.config_local").local_conf()
-local http               = require("resty.http")
+local socket_http        = require("socket.http")
 local core               = require("apisix.core")
```
Comment (soulbird) [inline apisix/discovery/eureka/init.lua:20]: We don't need to use blocking sockets at every phase, you can take a look at https://github.com/apache/apisix/blob/master/apisix/patch.lua and make some adjustments

**Comment by tzssangglass** [comment]
In the init and init_worker phases, we can't use `lua-resty-http`, so we use `luasocket`.
But we don't need to replace the runtime procedure that uses `lua-resty-http`.

**Comment by kongjun01** [comment]
> In the init and init_worker phases, we can't use `lua-resty-http`, so we use `luasocket`. But we don't need to replace the runtime procedure that uses `lua-resty-http`.

This method is public at init and runtime.

File: apisix/discovery/eureka/init.lua (hunk 1 above)
Comment (kongjun01) [reply apisix/discovery/eureka/init.lua:20]: > We don't need to use blocking sockets at every phase, you can take a look at https://github.com/apache/apisix/blob/master/apisix/patch.lua and make some adjustments

Ok，let me try.

**Comment by tzssangglass** [comment]
> This method is public at init and runtime.

I don't know what this means.

But replacing all `lua-resty.http` with `luasocket` is not acceptable. `luasocket` is a blocking call.

**Pull request closed without merging**